test/distributed/test_dist_loader.py

```diff
@@ -0,0 +1,131 @@
+import socket
+
+import pytest
+import torch
+import torch.multiprocessing as mp
+
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.datasets import FakeDataset, FakeHeteroDataset
+from torch_geometric.distributed import (
+    LocalFeatureStore,
+    LocalGraphStore,
+    Partitioner,
+)
+from torch_geometric.distributed.dist_context import DistContext, DistRole
+from torch_geometric.distributed.dist_neighbor_loader import DistNeighborLoader
+from torch_geometric.distributed.dist_neighbor_sampler import (
+    DistNeighborSampler,
```

Comment:
Mh, how shall we merge this if `DistNeighborSampler` does nto yet exist?

Reply:
This is probably the most difficult one to answer. I see two options: we might do a coordinated force merge, with prior check on unit tests with the complete stack; option 2 would be to close this draft and open a new PR that will contain all new distributed files.